model_compression_toolkit/core/keras/hessian/activation_hessian_calculator_keras.py

```diff
@@ -0,0 +1,361 @@
+from typing import List, Tuple, Dict, Any
+
+import tensorflow as tf
+from tensorflow.python.keras.engine.base_layer import Layer
+from tqdm import tqdm
+import numpy as np
+
+from model_compression_toolkit.constants import MIN_JACOBIANS_ITER, JACOBIANS_COMP_TOLERANCE, EPS
+from model_compression_toolkit.core.common.graph.edge import EDGE_SINK_INDEX
+from model_compression_toolkit.core.common import Graph, BaseNode
+from model_compression_toolkit.core.common.graph.functional_node import FunctionalNode
+from model_compression_toolkit.core.common.hessian.hessian_config import HessianConfig, HessianMode, HessianGranularity
+from model_compression_toolkit.core.keras.back2framework.instance_builder import OperationHandler
+from model_compression_toolkit.core.keras.hessian.hessian_calculator_keras import HessianCalculatorKeras
+from model_compression_toolkit.logger import Logger
+from tensorflow.python.util.object_identity import Reference as TFReference
+
+
+class ActivationHessianCalculatorKeras(HessianCalculatorKeras):
+    """
+    Hessian w.r.t activation for Keras graph computation.
+    """
+
+    def __init__(self,
+                 graph: Graph,
+                 config: HessianConfig,
+                 input_images: List[tf.Tensor],
+                 fw_impl):
+        """
+
+        Args:
+            graph: Float graph to compute its hessian data.
+            config: HessianConfig to use for during Hessian computation.
+            input_images: List of images to use the the computaion (image per graph input).
+            fw_impl: Framework implementation to use during computation.
+        """
+
+        super(ActivationHessianCalculatorKeras, self).__init__(graph=graph,
+                                                               config=config,
+                                                               input_images=input_images,
+                                                               fw_impl=fw_impl)
+
+    def compute(self) -> Dict[BaseNode, float]:
+        """
+        Compute the hessian of the float graph based on the configuration and images that in
+        the calculator.
+
+        Returns: Dictionary from interest point to hessian score.
+
+        """
+        if self.config.granularity == HessianGranularity.PER_LAYER:
+            output_list = self._get_model_output_replacement()
+            all_outputs_indices = []
+            if self.config.search_output_replacement:
+                all_outputs_indices = self._update_ips_with_outputs_replacements(output_list,
```

Comment:
GPTQ doesn't do this output_replacement hack, I think this change would make it run this too, and I don't know what would it do to the GPTQ execution.
Because we are planning on removing it, maybe for now just add a flag to skip this part when calling from GPTQ.